workers/unity/Packages/com.improbable.gdk.tools/DevAuthTokenUtils.cs

```diff
@@ -33,18 +43,48 @@ private static void Generate()
 
             try
             {
-                if (Json.Deserialize(receivedMessage).TryGetValue(JsonDataKey, out var jsonData) &&
+                var deserializedMessage = Json.Deserialize(receivedMessage);
+                if (deserializedMessage.TryGetValue(JsonDataKey, out var jsonData) &&
                     ((Dictionary<string, object>) jsonData).TryGetValue(TokenSecretKey, out var tokenSecret))
                 {
                     devAuthToken = (string) tokenSecret;
                 }
+                else
+                {
+                    throw new Exception(
+                        $@"{(deserializedMessage.TryGetValue(ErrorKey, out var errorMessage)
+                            ? errorMessage
+                            : string.Empty)}");
+                }
             }
             catch (Exception e)
             {
                 Debug.LogError($"Unable to generate Dev Auth Token. {e.Message}");
-                return;
+                return false;
             }
 
+            Debug.Log($"Saving token {devAuthToken} to Editor Preferences.");
+            EditorPrefs.SetString(DevAuthTokenKey, devAuthToken);
```

Comment:
this is across all projects - so need to think about how to identify the right one for the project